src/java.base/share/classes/java/util/random/RandomGeneratorFactory.java

```diff
@@ -0,0 +1,524 @@
+     */
+    @SuppressWarnings("unchecked")
+    private Map<RandomGeneratorProperty, Object> getProperties() {
+        Map<RandomGeneratorProperty, Object> props = properties;
+        if (props == null) {
+            synchronized (provider) {
+                if (properties == null) {
+                    try {
+                        Method getProperties = provider.type().getDeclaredMethod("getProperties");
+                        PrivilegedExceptionAction<Map<RandomGeneratorProperty, Object>> getAction = () -> {
+                            getProperties.setAccessible(true);
+                            return (Map<RandomGeneratorProperty, Object>)getProperties.invoke(null);
+                        };
+                        props = properties = AccessController.doPrivileged(getAction);
+                    } catch (SecurityException | NoSuchMethodException | PrivilegedActionException ex) {
+                        props = properties = Map.of();
+                    }
+                }
+            }
+        }
+
+        return props;
+    }
+
+    private Object getProperty(RandomGeneratorProperty property, Object defaultValue) {
+        return getProperties().getOrDefault(property, defaultValue);
+    }
+
+    /**
+     * Return true if the provider is a subclass of the category.
+     *
+     * @param category Interface category, sub-interface of {@link RandomGenerator}.
+     *
+     * @return true if the provider is a subclass of the category.
+     */
+    private boolean isSubclass(Class<? extends RandomGenerator> category) {
+        return isSubclass(category, provider);
+    }
+
+    /**
+     * Return true if the provider is a subclass of the category.
+     *
+     * @param category Interface category, sub-interface of {@link RandomGenerator}.
+     * @param provider Provider that is being filtered.
+     *
+     * @return true if the provider is a subclass of the category.
+     */
+    private static boolean isSubclass(Class<? extends RandomGenerator> category,
+                                      Provider<? extends RandomGenerator> provider) {
+        return provider != null && category.isAssignableFrom(provider.type());
+    }
+
+    /**
+     * Returns the provider matching name and category.
+     *
+     * @param name      Name of RandomGenerator
+     * @param category  Interface category, sub-interface of {@link RandomGenerator}.
+     *
+     * @return A provider matching name and category.
+     *
+     * @throws IllegalArgumentException if provider is not a subclass of category.
+     */
+    private static Provider<? extends RandomGenerator> findProvider(String name,
+                                                                    Class<? extends RandomGenerator> category)
+            throws IllegalArgumentException {
+        Map<String, Provider<? extends RandomGenerator>> fm = getFactoryMap();
+        Provider<? extends RandomGenerator> provider = fm.get(name.toUpperCase());
```

Comment:
again use of toUpperCase() instead of toUpperCase(Locale)

Reply:
removed